src/core/server/saved_objects/service/lib/aggs_utils.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexMapping } from '../../mappings';
+
+import { SavedObjectsErrorHelpers } from './errors';
+import { hasFilterKeyError } from './filter_utils';
+import { SavedObjectsAggs, validateSavedObjectTypeAggs } from './saved_objects_aggs_types';
+
+export const validateGetSavedObjectsAggs = (
+  allowedTypes: string[],
+  aggs: SavedObjectsAggs,
+  indexMapping: IndexMapping
+): SavedObjectsAggs => {
+  validateSavedObjectTypeAggs(aggs);
+  return validateGetAggFieldValue(allowedTypes, aggs, indexMapping);
+};
+
+const validateGetAggFieldValue = (
+  allowedTypes: string[],
+  aggs: any,
+  indexMapping: IndexMapping
+): SavedObjectsAggs => {
+  return Object.keys(aggs).reduce((acc, key) => {
+    if (key === 'field') {
+      const error = hasFilterKeyError(aggs[key], allowedTypes, indexMapping);
```

Comment:
I was going to comment, then I saw https://github.com/elastic/kibana/pull/64002/files#r412809210.

> We would just have to educate developers that they always need to use this alternative syntax which might not be obvious, but I don't think that's a problem.

I hear the explanation, and even if I agree that supporting all syntaxes will be a pain to express in term of types and validation, by enforcing a specific syntax, we are going to have a LOT of feedback from developers

For instance, from the ES official doc, this is the example for `filter`

```ts
filter: {
    term: {
        'field': 'value',
    },
},
```

which fails our aggs types validation.